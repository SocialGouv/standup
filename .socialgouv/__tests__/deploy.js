import util from "util"
import { directory } from "tempy"
import { project } from "@socialgouv/kosko-charts/testing/fake/github-actions.env"

const exec = util.promisify(require('child_process').exec);

jest.setTimeout(1000 * 60)

test("deploy dev 2", async () => {
  const dir = directory()
  const env = project("standup").dev

  env.SOCIALGOUV_CONFIG_PATH = `${dir}/autodevops/config.json`

  Object.assign(process.env, env)

  const cmd = `
    npx degit SocialGouv/kosko-charts/templates/autodevops ${dir}/autodevops; \
    yarn --cwd ${dir}/autodevops --silent; \
    cp -r ${__dirname}/../environments ${__dirname}/../config.json ${dir}/autodevops/; \
    yarn --cwd ${dir}/autodevops --silent generate --env dev
  `

  const { stdout: manifest } = await exec(cmd, { env: process.env })

  expect(manifest).toMatchSnapshot()
})
