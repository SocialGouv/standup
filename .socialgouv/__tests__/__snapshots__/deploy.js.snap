// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deploy dev 2 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: ''
    git/branch: refs/heads/mybranch
    git/remote: socialgouv/standup
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/standup
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    azure-pg-admin-user: standup
    application: standup
    owner: standup
    team: standup
    cert: wildcard
  name: standup-mybranch
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/standup
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: standup
    application: standup
    owner: standup
    team: standup
    cert: wildcard
    component: nginx
  name: standup
  namespace: standup-mybranch
spec:
  replicas: 1
  selector:
    matchLabels:
      app: standup
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/standup
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: standup
        application: standup
        owner: standup
        team: standup
        cert: wildcard
        component: nginx
    spec:
      containers:
        - image: ghcr.io/socialgouv/fabrique/standup:undefined
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: standup
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 32Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /index.html
              port: http
            periodSeconds: 5
          envFrom:
            - secretRef:
                name: standup-sealed-secret
            - configMapRef:
                name: standup-configmap
---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  annotations:
    sealedsecrets.bitnami.com/cluster-wide: 'true'
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/standup
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  name: standup-sealed-secret
  namespace: standup-mybranch
  labels:
    application: standup
    owner: standup
    team: standup
    cert: wildcard
spec:
  encryptedData:
    SENTRY_DSN: >-
      AgBApFjJq6o6w6lQA3/6LX9Hk7EhIQgz6y/uBZh7UegeYq/vtjmXjbbOJsNna2x30E6DCUMWNbqVne1lmgJaPJn0t4u5k0bs8pW7GLQwpLmRtE6005b+bjyqH4QNRCEriaOyQrjQldxW7/dqBQRLD1RCh6AGSacDxbEh7Dw8WaeBIvdq64tu1w4RVVCkjekLUR4MRjaTbK8QDYyHWGW3szHcrWGiV3MpLeNi+dXTzWPuVcWocRmjLQOTwv7Issfcj4A+GjoRpOherF0cYFlpO+nEPtDh+lr9lOTsTdWilb+woWCa9QrYxjuPBXVFosmKpUHeafpIKqhIAP2MGkcpBNxNAdwWB1cFsImgWNAl6883bBJ8FdiOn6BViZlyYD0fWRzVjNy39X0JgKeuQKvRkMpqbSrLGZts5ElPBzT458e3v/M/2mcIKXAtjTWxCYpmIqYP3HI90yBIYQYErYu5sbxKxUFpW6Bq+vwINxkuFu+kyIUmDDC+gjzTes1sf8D2QWIvXnvy87w946+1fnbzZuFlZzG6xf7DwomC1KRp0BFpxa3sPrcDdyGkvPNq9JTWmEpcUyJq9SEZGC0+hRiffBo+B5UeN7pCxQrTViIZZBEb54v61VOu7MSNH3gVQnR4aiAFgite5GAqxOqNaYybglef/WUdm4wL6C++jB2E5cNSpz3JWTk22sDfCvgRNcHbW/MpiQzRIBOrpGFqEHpE/TZ6JIplSxHZ1UF440hTOVLpO2PWSHToVQ1r3iY9/M6NcOtKdHdEShYrWSak0feMYlvN2ofXtXfbUlUbQw==
  template:
    metadata:
      annotations:
        sealedsecrets.bitnami.com/cluster-wide: 'true'
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/standup
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      name: standup-sealed-secret
      labels:
        application: standup
        owner: standup
        team: standup
        cert: wildcard
    type: Opaque
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: standup-configmap
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/standup
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    application: standup
    owner: standup
    team: standup
    cert: wildcard
  namespace: standup-mybranch
data:
  NODE_ENV: production
  GITHUB_ORGANIZATION: SocialGouv
  HASURA_URL: https://hasura.carnets.fabrique.social.gouv.fr/v1/graphql
  MATOMO_SITE_ID: '30'
  MATOMO_URL: https://matomo.fabrique.social.gouv.fr/
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: standup
    application: standup
    owner: standup
    team: standup
    cert: wildcard
  name: standup
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/standup
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: standup-mybranch
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: standup
  type: ClusterIP
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/standup
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: standup
    application: standup
    owner: standup
    team: standup
    cert: wildcard
  name: standup
  namespace: standup-mybranch
spec:
  rules:
    - host: standup-mybranch.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              serviceName: standup
              servicePort: 80
            path: /
  tls:
    - hosts:
        - standup-mybranch.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
"
`;
